### DISCUSSION TITLE
Why are all the notifications closed as group?

### DISCUSSION THREAD

**Discussion by lordofscripts** [dunst-project/dunst#1359] [Help]
My Dunst server relevant configuration is:

- **Left** mouse: `do_close`
- **Middle** mouse: `do_action, do_close`
- **Right** mouse: `do_close_all`

In my GO program I use a Private DBus application session with these:

- Notification expiry up to server
- Open/Cancel **actions** defined for each notification.
- Listeners for `onClose` & `onAction` setup.
- Fire 3 separate notifications
- My code **does not** close notification programmatically. I do any of these:

- Let them expire in a few seconds (up to the server), or
- Left click on ONE to close, or
- Middle click to invoke the action

I noticed that if I fired several notifications with the same connection, **Left Mouse** (close ONE) would close **all notifications** that were fired under the same connection. 

However, If I used **one connection for each notification**, I could _close them individually_ as per the Dunst configuration (left mouse click).

So, having observed that, I wanted to do the same exercise of _firing several, each with a dedicated connection_, but with `onAction` and `onClose` listeners. The _connection being kept alive until_ **Close** arrived. So far, so good, alright?

To my surprise, in the 2nd scenario, when I **middle click** on **any** of the 3 notifications, **all three get closed simultaneously** (yes, way before the expiration in case anyone wonders). However, **Left click** acts on them individually.

So, why is this behaviour not consistent?❓

Remarks:
- I have not figured out how I could (programmatically!) specify the **default action**. Is it via a hint? How? Although the documentation mentions it, it doesn't say HOW.
- My system has DMenu installed and configured (default config) as action handler but I don't see context menus appearing anywhere.

**Comment by bynect** [comment]
> * I have not figured out how I could (programmatically!) specify the **default action**. Is it via a hint? How? Although the documentation mentions it, it doesn't say HOW.

it has to be called default


Anyway, the problem might be related to the rather old version of dunst you said you were using. Maybe you could try compiling from source the newer version? (it is not difficult)

**Reply by lordofscripts** [reply]
Thanks for the default action help!

Yes, I think it might be related to my old version. I believe last week I found, by chance, a discussion about such problem in v1.5.

I tried to recompile but, while it appears easy to someone who builds it regularly, I couldn't figure out how to properly configure it. It was also asking for a lot of header files I don't have, primarily due to the limited space & resources on my Raspberry π 2, which is a lightweight system and doesn't have a full-blown Gnome either. Mine is X11-based.

**Reply by bynect** [reply]
You can try disabling certain features, for example WAYLAND=0 before using make

Like so

WAYLAND=0 make dunst

To install the required packges see the section in the github wiki about compiling & dependencies
